packages/react-router/lib/dom/global.ts

```diff
@@ -1,22 +1,12 @@
 import type { HydrationState, Router as DataRouter } from "../router/router";
-import type { AssetsManifest, CriticalCss, FutureConfig } from "./ssr/entry";
+import type { ServerHandoff } from "../server-runtime/serverHandoff";
+import type { AssetsManifest } from "./ssr/entry";
 import type { RouteModules } from "./ssr/routeModules";
 
-export type WindowReactRouterContext = {
-  basename?: string;
-  state: HydrationState;
-  criticalCss?: CriticalCss;
-  future: FutureConfig;
-  ssr: boolean;
-  isSpaMode: boolean;
+export type WindowReactRouterContext = ServerHandoff & {
+  state: HydrationState; // Deserialized via the stream
   stream: ReadableStream<Uint8Array> | undefined;
   streamController: ReadableStreamDefaultController<Uint8Array>;
-  // The number of active deferred keys rendered on the server
-  a?: number;
-  dev?: {
-    port?: number;
-    hmrRuntime?: string;
-  };
```

Comment:
Unused in the repo.  I believe `a` was from the `defer` implementation and I think the `dev` stuff was from the classic compiler?